### DISCUSSION TITLE
Performance / Speed analysis

### DISCUSSION THREAD

**Discussion by hbinkle** [JanDeDobbeleer/oh-my-posh#797] [Troubleshoot] [answered]
Hi,
my powershell is sometimes very slow. It's caused by the git or kubectl segment.
Is there any way to get some debug output how long a segment needs to evaluated?
That way it would be very easy to determine any turn off the segment.

Or even better: how about the option to define timeouts for segments data retrieval and refresh times?
e.g. for kubectl I'd like set a refresh time of 3 min and have a command to manually force a refesh.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@hbinkle you can make use of `Write-PoshDebug` as is listed [here](https://ohmyposh.dev/docs/faq#the-prompt-is-slow-delay-in-showing-the-prompt-between-commands).
